extensions/amp-story/1.0/amp-story-form.js

```diff
@@ -37,116 +18,84 @@ export function allowlistFormActions(win) {
-  let errorEl = scopedQuerySelector(formEl, AttributeElementSelector.ERROR);
+let FormElementChildrenDef;
 
-  // Create and append fallback form attribute elements, if necessary.
-  if (!submittingEl) {
-    submittingEl = createFormSubmittingEl_(formEl);
-    formEl.appendChild(submittingEl);
-  }
-  if (!successEl) {
-    successEl = createFormResultEl_(win, formEl, true);
-    formEl.appendChild(successEl);
-  }
-  if (!errorEl) {
-    errorEl = createFormResultEl_(win, formEl, false);
-    formEl.appendChild(errorEl);
-  }
-}
+/**
+ * @const {Object<string, (function(Element):!FormElementChildrenDef|function():!FormElementChildrenDef)>}
+ */
+const createStatusChildrenByAttribute = {
+  'submitting': () => {
+    const loadingSpinner = new LoadingSpinner();
+    const element = loadingSpinner.build();
+    loadingSpinner.toggle(true /* isActive */);
+    return element;
+  },
+
+  'submit-success': (formEl) =>
+    createFormResultChildren(
+      localize(formEl, LocalizedStringId.AMP_STORY_FORM_SUBMIT_SUCCESS)
+    ),
+
+  'submit-error': (formEl) =>
+    createFormResultChildren(
+      localize(formEl, LocalizedStringId.AMP_STORY_FORM_SUBMIT_ERROR)
+    ),
+};
 
 /**
- * @param {!Element} formEl The form to which the attribute elements belong.
- * @return {!Array<!Element>} The list of response attribute elements that
- *     belong to the given form.
- * @private
+ * @param {Element} parent
+ * @param {string} attr
+ * @return {?Element}
  */
-export function getResponseAttributeElements(formEl) {
-  const selector = `
-    ${AttributeElementSelector.SUBMITTING},
-    ${AttributeElementSelector.SUCCESS},
-    ${AttributeElementSelector.ERROR}`;
-  return Array.from(scopedQuerySelectorAll(formEl, selector));
+function selectByAttr(parent, attr) {
+  return parent.querySelector(`[${escapeCssSelectorIdent(attr)}]`);
 }
 
 /**
- * Create an element that is used to display the in-progress state of a form
- * submission attempt.
- * @param {!Element} formEl The form to which the `submitting` element will be
+ * Add a default form attribute element for each absent response attribute.
+ * @param {!Element} formEl The form to which the attribute elements will be
  *     added.
- * @return {!Element}
  * @private
  */
-function createFormSubmittingEl_(formEl) {
-  const submittingEl = createResponseAttributeEl_(
-    formEl,
-    FormResponseAttribute.SUBMITTING
-  );
-  const loadingSpinner = new LoadingSpinner();
-  submittingEl.firstElementChild.appendChild(loadingSpinner.build());
-  loadingSpinner.toggle(true /* isActive */);
-  return submittingEl;
+export function setupResponseAttributeElements(formEl) {
+  for (const attribute in createStatusChildrenByAttribute) {
+    const el = selectByAttr(formEl, attribute);
+    if (!el) {
+      const children = createStatusChildrenByAttribute[attribute](formEl);
+      const element = (
+        <div {...{[attribute]: true}}>
```

Comment:
This will conflict with https://github.com/ampproject/amphtml/pull/36700, which I'd like to merge for security concerns.

Reply:
(It's also not fully dynamic since the value determined within source.)

Reply:
With this having such a small node, I would just do a set attribute? We could try allowing spread of inline object expressions, since they keys would have to be provided manually (and it's not as dynamic as _any_ object referenced by an identifier). But that still wouldn't cover this case.

Maybe hoist this `<div class="i-amphtml-story-page-attachment-form-submission-status">` into a component, and rewrite the `createStatus` helpers to do:

```js
const const createStatusChildrenByAttribute = {
  'submitting': () => {
    return <div submitting>
      <Status>
        {children}
      </Status>
    </div>;
  }
}
```